geometry/eolearn/geometry/transformations.py

```diff
@@ -257,26 +294,49 @@ def execute(self, eopatch):
         :return: New EOPatch with vector data transformed into a raster feature
         :rtype: EOPatch
         """
+
         if eopatch.bbox is None:
             raise ValueError('EOPatch has to have a bounding box')
 
         height, width = self._get_raster_shape(eopatch)
 
         group_classes = self.overlap_value is not None
-        rasterization_shapes = self._get_rasterization_shapes(eopatch, group_classes=group_classes)
+        
+        affine_transform = rasterio.transform.from_bounds(*eopatch.bbox, width=width, height=height)
+        rasterize_args = dict(self.rasterio_params, transform=affine_transform, dtype=self.raster_dtype)
+        rasterize_func = rasterio.features.rasterize if self.overlap_value is None else self.rasterize_overlapped
 
-        if not rasterization_shapes:
-            eopatch[self.raster_feature] = np.full((height, width, 1), self.no_data_value, dtype=self.raster_dtype)
+        if self.raster_feature[0].is_timeless():
+            rasterization_shapes = self._get_rasterization_shapes(eopatch, group_classes=group_classes)
+    
+            if not rasterization_shapes:
+                eopatch[self.raster_feature] = np.full((height, width, 1), self.no_data_value, dtype=self.raster_dtype)
+                return eopatch
+    
+            raster = self._get_raster(eopatch, height, width)
+            rasterize_func(rasterization_shapes, out=raster, **rasterize_args)
+    
+            eopatch[self.raster_feature] = raster[..., np.newaxis]
             return eopatch
+        
+        else:   
+            timestamp_flags = self._sort_timestamps(eopatch)
+            timed_raster = np.full((len(timestamp_flags), height, width), self.no_data_value, dtype=self.raster_dtype)            
+            
+            #loop over timestamps
+            for idx, time, flag in timestamp_flags:
+                if flag==1:
+                    rasterization_shapes = self._get_rasterization_shapes(eopatch, group_classes=group_classes, timestamp=time)
 
-        affine_transform = rasterio.transform.from_bounds(*eopatch.bbox, width=width, height=height)
-        rasterize_args = dict(self.rasterio_params, transform=affine_transform, dtype=self.raster_dtype)
+                    if not rasterization_shapes:
+                        break
 
-        raster = self._get_raster(eopatch, height, width)
-        rasterize_func = rasterio.features.rasterize if self.overlap_value is None else self.rasterize_overlapped
-        rasterize_func(rasterization_shapes, out=raster, **rasterize_args)
+                    raster = self._get_raster(eopatch, height, width)
```

Comment:
`_get_raster` method should be called only once and should return what is currently called `timed_raster`. Only this way it will be possible to rasterize more things into an existing raster feature. At this point, you can just slice the full temporal raster.

Reply:
I updated `_get_raster` so it now is only called once and returns the either the existing raster feature, an array with no time dimension, or an array with a time dimension depending on the case.